huaweicloud/services/apig/resource_huaweicloud_apig_application_authorization.go

```diff
@@ -0,0 +1,328 @@
+		DeleteContext: resourceAppAuthDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAppAuthImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(3 * time.Minute),
+			Update: schema.DefaultTimeout(3 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the application and APPCODEs are located.",
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the dedicated instance to which the application and APIs belong.",
+			},
+			"application_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the application authorized to access the APIs.",
+			},
+			"env_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The environment ID where the APIs were published.",
+			},
+			"api_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The authorized API IDs",
+			},
+		},
+	}
+}
+
+func resourceAppAuthCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+	var (
+		appId  = d.Get("application_id").(string)
+		envId  = d.Get("env_id").(string)
+		apiIds = utils.ExpandToStringListBySet(d.Get("api_ids").(*schema.Set))
+	)
+	err = createAppAuthForApis(ctx, client, d, apiIds, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(fmt.Sprintf("%s/%s", envId, appId))
+
+	return resourceAppAuthRead(ctx, d, meta)
+}
+
+func flattenAuthorizedApis(apiInfos []appauths.ApiAuthInfo) []string {
+	result := make([]string, len(apiInfos))
+	for i, val := range apiInfos {
+		result[i] = val.ApiId
+	}
+	return result
+}
+
+func resourceAppAuthRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	var (
+		instanceId = d.Get("instance_id").(string)
+		appId      = d.Get("application_id").(string)
+		opts       = appauths.ListOpts{
+			InstanceId: instanceId,
+			AppId:      appId,
+		}
+	)
+	resp, err := appauths.ListAuthorized(client, opts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, fmt.Sprintf("error querying authorized APIs from application (%s) under dedicated instance (%s)",
```

Comment:
I think here can just return diag.Errorf("")

Reply:
if the instance or application is not exist, this request will return 404 error, not an empty list.